### DISCUSSION TITLE
Pausing MITM from proxying requests

### DISCUSSION THREAD

**Discussion by borisspektor** [mitmproxy/mitmproxy#7210] [Q&A] [answered]
Sometimes when using the proxy some apps and features fail to work correctly. I usually fix this by using ignore_hosts, but sometimes I would like to completely "ignore all hosts" or "pause" the proxy, to allow all requests to be sent directly.
How can I achieve that? I am using mitmweb but I would like to somehow send an http request to the proxy to enable/disable this behavior.

**Comment by emanuele-em** [comment]
HI @borisspektor! `--ignore_hosts` accepts also regex and that could be useful for what you need 😄 [--ignore_hosts doc](https://docs.mitmproxy.org/stable/concepts-options/#ignore_hosts)

you can run mitmproxy (mitmweb is perfect too) with this command:
```bash
mitmweb --set ignore_hosts='^.*$'
```

I never tested this, I hope it works 👍🏼

**Reply by borisspektor** [reply]
@emanuele-em using `ignore_hosts` like this does indeed work, but it requires a "restart" of mitmproxy. I would like to be able to set `ignore_hosts` while the proxy is running - in order to keep the already captured data. Stopping and running again would result in losing this data and starting over.

**Reply by emanuele-em** [reply]
@borisspektor actually you should be able to change `ignore_hosts` option directly inside the UI without stopping mitmweb:

<img width="1500" alt="image" src="https://github.com/user-attachments/assets/73de9d5e-9382-4074-bfbe-472cd2063d20">

you can open this modal by clicking the top-left options button inside the options tab:

<img width="455" alt="image" src="https://github.com/user-attachments/assets/a071b2be-1c32-4519-9fa6-c4a3bc01e59f">

**Reply by borisspektor** [reply]
@emanuele-em looks like this option using the UI is not working unfortunately... does it work for you?

**Comment by rosydawn6** [comment, marked as answer]
I modify options without a restart like this:

`  ctx.options.update(mode=list(tempList))`

I have not tried for ctx.options.ignore_hosts but it should work in the same way:

`  ctx.options.update(ignore_hosts=list(tempList))`

**Reply by borisspektor** [reply]
@rosydawn6 thanks, looks like it's working :)
